packages/@blockprotocol/graph/src/stdlib/subgraph/element/entity.ts

```diff
@@ -0,0 +1,124 @@
+import { Entity, EntityId, EntityVersion } from "../../../types/entity.js";
+import { Subgraph } from "../../../types/subgraph.js";
+import { Timestamp } from "../../../types/subgraph/time.js";
+import { isEntityVertex } from "../../../types/subgraph/vertices.js";
+
+/**
+ * Returns all `Entity`s within the vertices of the subgraph, optionally filtering to only get their latest editions.
+ *
+ * @param subgraph
+ * @param latest - whether or not to only return the latest editions of each entity
+ */
+export const getEntities = (
+  subgraph: Subgraph,
+  latest: boolean = false,
+): Entity[] => {
+  return Object.values(
+    Object.values(subgraph.vertices).flatMap((versionObject) => {
+      const entityEditionVertices = latest
+        ? Object.keys(versionObject)
+            .sort()
+            .slice(-1)
+            .map((latestVersion) => versionObject[latestVersion]!)
+            .filter(isEntityVertex)
+        : Object.values(versionObject).filter(isEntityVertex);
+
+      return entityEditionVertices.map((vertex) => vertex.inner);
+    }),
+  );
+};
+
+/**
+ * Gets an `Entity` by its `EntityId` from within the vertices of the subgraph. If `targetEditionInformation` is not passed,
+ * then the latest version of the `Entity` will be returned.
+ *
+ * Returns `undefined` if the entity couldn't be found.
+ *
+ * @param subgraph
+ * @param {EntityId} entityId - The `EntityId` of the entity to get.
+ * @param {EntityVersion|Timestamp} [targetEditionInformation] - Optional information needed to uniquely identify an edition of
+ *     an entity either by an explicit `EntityVersion`, or by a given `Timestamp` where the entity whose lifespan
+ *     overlaps the given timestamp will be returned.
+ * @throws if the vertex isn't an `EntityVertex`
+ */
+export const getEntity = (
+  subgraph: Subgraph,
+  entityId: EntityId,
+  targetEditionInformation?: EntityVersion | Timestamp | Date,
+): Entity | undefined => {
+  const entityEditions = subgraph.vertices[entityId];
+
+  if (entityEditions === undefined) {
+    return undefined;
+  }
+
+  const editionVersions = Object.keys(entityEditions).sort();
+
+  let entityEdition: Entity | undefined;
+
+  // Short circuit for efficiency, just take the latest
+  if (targetEditionInformation === undefined) {
+    [entityEdition] = editionVersions
+      .slice(-1)
+      .map((latestVersion) => entityEditions[latestVersion]!.inner);
```

Comment:
Alternative if mutating the original array isn't a concern, which I find a bit easier to parse
```suggestion
    entityEdition = entityEditions[edititionVersions.pop()]!.inner;
```

Reply:
This behaves differently though, the pop will throw an error whereas the map will return `undefined`

Reply:
Oh wait it doesn't throw, let me check if I'm happy with it being mutated

Reply:
That syntax doesn't work anyway because pop can be undefined so we'd need a temporary variable and such. Going to leave it as is
<img width="831" alt="image" src="https://user-images.githubusercontent.com/25749103/207151261-ead2f2bc-5b50-467c-9f46-c0a19c3fee5e.png">

Reply:
Fair enough